server/src/main/java/org/elasticsearch/operator/OperatorHandler.java

```diff
@@ -0,0 +1,118 @@
+     * {@link TransformState}, which contains the current cluster state as well as any previous keys
+     * set by this handler on prior invocation.
+     *
+     * @param source The parsed information specific to this handler from the combined cluster state content
+     * @param prevState The previous cluster state and keys set by this handler (if any)
+     * @return The modified state and the current keys set by this handler
+     * @throws Exception
+     */
+    TransformState transform(Object source, TransformState prevState) throws Exception;
+
+    /**
+     * Sometimes certain parts of the cluster state cannot be created/updated without previously
+     * setting other cluster state components, e.g. composable templates. Since the cluster state handlers
+     * are processed in random order by the OperatorClusterStateController, this method gives an opportunity
+     * to any operator handler to declare other operator handlers it depends on. Given dependencies exist,
+     * the OperatorClusterStateController will order those handlers such that the handlers that are dependent
+     * on are processed first.
+     *
+     * @return a collection of operator handler names
+     */
+    default Collection<String> dependencies() {
+        return Collections.emptyList();
+    }
+
+    /**
+     * All implementations of OperatorHandler should call the request validate method, by calling this default
+     * implementation. To aid in any special validation logic that may need to be implemented by the operator handler
+     * we provide this convenience method.
+     *
+     * @param request the master node request that we base this operator handler on
+     */
+    default void validate(MasterNodeRequest<?> request) {
+        ActionRequestValidationException exception = request.validate();
+        if (exception != null) {
+            throw new IllegalStateException("Validation error", exception);
+        }
+    }
+
+    /**
+     * Convenience method to convert the incoming passed in input to the transform method into a map.
+     *
+     * @param input the input passed into the operator handler after parsing the content
+     * @return
+     */
+    @SuppressWarnings("unchecked")
+    default Map<String, ?> asMap(Object input) {
+        if (input instanceof Map<?, ?> source) {
+            return (Map<String, Object>) source;
+        }
+        throw new IllegalStateException("Unsupported " + name() + " request format");
+    }
+
+    /**
+     * Convenience method that creates a {@link XContentParser} from a content map so that it can be passed to
+     * existing REST based code for input parsing.
+     *
+     * @param config XContentParserConfiguration for this mapper
+     * @param source the operator content as a map
+     * @return
+     */
+    default XContentParser mapToXContentParser(XContentParserConfiguration config, Map<String, ?> source) {
```

Comment:
This doesn't seem to have anything to do with operator config, could it be in a more general place for xcontent utilities?